forwarder/pkg/common/egress_interface.go

```diff
@@ -78,25 +78,36 @@ func parseProcFile(scanner *bufio.Scanner) (string, net.IP, error) {
 		if strings.TrimSpace(parts[1]) == "00000000" {
 			outgoingInterface := strings.TrimSpace(parts[0])
 			defaultGateway := strings.TrimSpace(parts[2])
-			ip := parseGatewayIP(defaultGateway)
+			ip, err := parseGatewayIP(defaultGateway)
+			if err != nil {
+				logrus.Errorf("error processing gateway IP %v for outgoing interface: %v", defaultGateway, outgoingInterface)
+				break
```

Comment:
I see some problems in method `parseProcFile`
1) `len` of parts is not checked at line 77. We naively expect that has len >= 3
2) we can use `return` instead of a` break` at lines 70, 84 and return an actual error. Also, we can wrap it. For example `errors.Wrap(err, "failed to locate default route...")`

Reply:
Thanks for the pointers @denis-tingajkin updated.